libcudacxx/include/cuda/std/detail/libcxx/include/optional

```diff
@@ -264,6 +264,7 @@ struct __optional_destruct_base<_Tp, false>
     };
     bool __engaged_;
 
+    _CCCL_EXEC_CHECK_DISABLE
```

Comment:
I need this to allow `cuda::std::optional` to destruct an `nvtx3::scoped_range_in`. @miscco is aware and may provide a more generic PR in-time.